packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -44,6 +44,16 @@ export type InsightsProps<
    * @internal indicator for the default insights middleware
    */
   $$internal?: boolean;
+  /**
+   * If this is true, don't send events, unless the search response contains:
+   *
+   * {
+   *   "renderingContent": {
+   *     "analytics": true
+   *  }
+   * }
+   */
+  verifyEventPermission?: boolean;
```

Comment:
is this the right name for the option? (it's essentially to detect "insights is not set to true by the customer")